docs/manual/howto/widget.rst

```diff
@@ -433,6 +433,63 @@ For example, the Mpris2 widget uses the following code:
 interfaces. Refer to the `dbus-next documentation <https://python-dbus-next.readthedocs.io/en/latest/>`_
 for more information on how to use the module.
 
+Mouse events
+============
+
+By default, widgets handle button presses and will call any function that is bound to the button in the
+``mouse_callbacks`` dictionary. The dictionary keys are as follows:
+
+ - ``Button1``: Left click
+ - ``Button2``: Middle click
+ - ``Button3``: Right click
+ - ``Button4``: Scroll up
+ - ``Button5``: Scroll down
+ - ``Button6``: Scroll left
+ - ``Button7``: Scroll right
+
+You can then define your button bindings in your widget (e.g. in ``__init__``):
+
+.. code:: python
+
+    class MyWidget(widget.TextBox)
+
+        def __init__(self, *args, **config):
+            widget.TextBox.__init__(self, *args, **kwargs)
+            self.add_callbacks(
+                {
+                    "Button1": self.left_click_method,
+                    "Button3": self.right_click_method
+                }
+            )
+
+In addition to button presses, you can also respond to mouse enter and leave events.
+For example, to make a clock show a longer date when you put your mouse over it, you
+can do the following:
+
+.. code:: python
+
+    class MouseOverClock(widget.Clock):
+        defaults = [
+            (
+                "long_format",
+                "%A %d %B %Y | %H:%M",
+                "Format to show when mouse is over widget."
+            )
+        ]
+
+        def __init__(self, **config):
+            widget.Clock.__init__(self, **config)
+            self.add_defaults(MouseOverClock.defaults)
+            self.short_format = self.format
+
+        def mouse_enter(self, *args, **kwargs):
+            self.format = self.long_format
+            self.bar.draw()
+
+        def mouse_leave(self, *args, **kwargs):
+            self.format = self.short_format
+            self.bar.draw()
+
```

Comment:
Perhaps add a comment to say that `self.bar.draw()` is needed only if the size of the widget changes, otherwise `self.draw()` is sufficient?

Reply:
That's already included further up the page.

Reply:
Ah, it's too early for my brain